frontend/ui/elements/logviewer.lua

```diff
@@ -0,0 +1,150 @@
+local Font = require("ui/font")
+local TextViewer = require("ui/widget/textviewer")
+local UIManager = require("ui/uimanager")
+local _ = require("gettext")
+local T = require("ffi/util").template
+
+local function downscale(font_size)
+    local min = 8
+    return font_size > min and font_size - 1 or min
+end
+
+local function upscale(font_size)
+    local max = 16
+    return font_size < max and font_size + 1 or max
+end
+
+local function checkFile(f)
+    if not f then return end
+    local file = io.open(f, "r")
+    if file then
+        file:close()
+        return true
+    end
+end
+
+local LogViewer = {}
+
+function LogViewer:new(o)
+    o.exists = checkFile(o.file)
+    setmetatable(o, self)
+    self.__index = self
+    return o
+end
+
+function LogViewer:savedLines()
```

Comment:
why "_saved_ lines" ? Do you mean "_tailed_ lines" ?

Reply:
what about `getTail()` and `setTail(lines)`

Reply:
The name "getTail()" sounds like it will return some content, not a number.
so, get/setNbLinesForTail or getNbTailLines ?

Reply:
`(g|s)etTailCount`?